### ISSUE TITLE
EditorConfig + term_start() makes cursor jump to the beginning of the buffer

### ISSUE THREAD

**Issue by mjyut** [vim/vim#15854]
### Steps to reproduce

1. Create a `.editorconfig` file in the current directory.
     This `.editorconfig` must have content.

    ```ini
    [*]
    charset = utf-8
    ```

2. Start Vim.

    ```sh
    vim -u NONE
    ```

3. Inside Vim, enable the built-in package EditorConfig.

    ```vim
    :packadd editorconfig
    ```

4. Enter something into the current buffer.

    ```text
    foo.
    bar.
    baz........
    ````

5. Place the cursor somewhere **other** than the **beginning of the line**.

    ```text
    foo.
    bar.
    baz........<Cursor is here>
    ````

6. Call `term_start()` with `hidden`  option set to true.

    ```vim
    :call term_start(['true'], #{hidden: 1, term_finish: 'close'})
    ```

7. Strangely, the cursor moves to the beginning of the buffer.

    ```text
    <Cursor jump to here>
    bar.
    baz........
    ````

---

This issue affects the FZF plugin.

### Expected behaviour

The cursor does not move on calling `term_start()`.

### Version of Vim

9.1 (patches: 1-774)

### Environment

OS: Linux
Terminal: urxvt (Same issue in GVim)

**Comment by chrisbra** [comment]
so has this something to do with the Editorconfig plugin?

**Comment by mjyut** [comment]
> so has this something to do with the Editorconfig plugin?

I have not been able to find any other way to reproduce this problem.
To reproduce this, both the EditorConfig plugin must be enabled and `.editorconfig` must exist.
Of course, it is possible that the problem is caused by a function that EditorConfig uses internally. It is beyond my ability to identify it.

**Comment by yegappan** [comment]
I am able to reproduce this issue without the editorconfig plugin using the following code:

```
func Test_Cursor()
    new
    call setline(1, ['one', 'two', 'three'])
    call cursor(3, 10)
    augroup TestCursor
	autocmd BufFilePost * call setbufvar(3, '&tabstop', 4)
    augroup END

    call term_start(['true'], #{hidden: 1, term_finish: 'close'})

    augroup TestCursor
	au!
    augroup END
endfunc
call Test_Cursor()
```

Place the above code in a file (test.vim) and source the file in a new instance of Vim.
The cursor should end up in the first line and column.

**Comment by chrisbra** [comment]
Seems we need to save the cursor when temporarily switching to the hidden terminal:
```diff
diff --git a/src/terminal.c b/src/terminal.c
index 073f8dd48..f61a54f7f 100644
--- a/src/terminal.c
+++ b/src/terminal.c
@@ -446,6 +446,7 @@ term_start(
     buf_T      *newbuf;
     int                vertical = opt->jo_vertical || (cmdmod.cmod_split & WSP_VERT);
     jobopt_T   orig_opt;  // only partly filled
+    pos_T      save_cursor;

     if (check_restricted() || check_secure())
        return NULL;
@@ -518,6 +519,7 @@ term_start(
        old_curbuf = curbuf;
        --curbuf->b_nwindows;
        curbuf = buf;
+       save_cursor = curwin->w_cursor;
        curwin->w_buffer = buf;
        ++curbuf->b_nwindows;
     }
@@ -763,6 +765,7 @@ term_start(
            --curbuf->b_nwindows;
            curbuf = old_curbuf;
            curwin->w_buffer = curbuf;
+           curwin->w_cursor = save_cursor;
            ++curbuf->b_nwindows;
        }
        else if (vgetc_busy
```

**Comment by yegappan** [comment]
@chrisbra Your patch fixes the problem.

**Comment by bfrg** [comment]
Is this related to the autocmd? Because there was already a very similar issue, see #5820 (fixed by Bram in 1e6bbfb5).

**Comment by chrisbra** [comment]
Yes and no. For the 
```
   call term_start(['true'], #{hidden: 1, term_finish: 'close'})
```
we do temporarily switch the current window to the terminal buffer and the autocommand verifies the cursor position for this window (which of course when starting up is empty and therefore the cursor is reset to the line 1 column 1) and the cursor remains there when switching back to the original buffer.
